Model/Transaction.php

```diff
@@ -386,8 +386,12 @@ protected function getProductTaxCode($item, $order)
                 $children = $item->getChildrenItems();
             }
 
-            if (!empty($children) && $children[0]->getProduct()->getTaxClassId()) {
-                $taxClass = $this->taxClassRepository->get($children[0]->getProduct()->getTaxClassId());
+            if (!empty($children) && is_array($children)) {
+                $child = array_first($children);
```

Comment:
Just so I understand (I don't write PHP often) -- where does `array_first` come from? My google-fu seems to only show this as a Laravel thing.

Reply:
@dbernheisel This is part of the `weew/helpers-array` package that's included with Magento.  
https://packagist.org/packages/weew/helpers-array

If you're worried about additional dependencies, we can rework it to use php's `reset` instead.  I chose `array_first` because it's a bit more intuitive.  
https://www.php.net/manual/en/function.reset.php

Reply:
@ryanreeves-taxjar Is the `weew/helpers-array` package still included in Magento 2.3+? Seeing it in 2.2 but not 2.3 or 2.4: https://devdocs.magento.com/guides/v2.4/release-notes/packages-open-source.html

I agree with using a native PHP method such as `reset` to avoid regressions in the future if Magento decides to remove a package from core.

Reply:
It looks like it was moved to the `magento2-functional-testing-framework` package, which is an optional dev dependency.  Updated to use `reset`.  

https://github.com/magento/magento2-functional-testing-framework/blob/develop/composer.json